src/Accord.Bot/Infrastructure/ReminderService.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Remora.Discord.API.Abstractions.Rest;
+using Remora.Discord.Core;
+
+namespace Accord.Services.Reminder
+{
+    public class ReminderService : BackgroundService
+    {
+        private readonly IServiceScopeFactory _serviceScopeFactory;
+
+        public ReminderService(IServiceScopeFactory serviceScopeFactory)
+        {
+            _serviceScopeFactory = serviceScopeFactory;
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            using var scope = _serviceScopeFactory.CreateScope();
+            var services = scope.ServiceProvider;
+
+            var mediator = services.GetRequiredService<IMediator>();
+
+            var channelApi = services.GetRequiredService<IDiscordRestChannelAPI>();
+
+            await ProcessReminders(mediator, channelApi, true, stoppingToken);
+
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                await ProcessReminders(mediator, channelApi, false, stoppingToken);
```

Comment:
I think a second is likely going to burn up a bit of CPU? Maybe every 15-30 seconds?

Reply:
it mostly reads out of cache, unless a new reminder is added/removed which it will need to invalidate the cache anyway. While i'm not sure about the actual CPU Usage impact that has, let's say TimeSpan.FromSeconds(15); is it fine?.
Also i'm guessing precision is not something the user goes after in that scenario, because i'm guessing the reminders will have a big timespan anyway.
On that note https://github.com/patrickklaeren/Accord/blob/f7c93675aed193c01a32e9e9cb36ae86b9f4987b/src/Accord.Bot/Infrastructure/ReminderService.cs#L47 should i increase this more than 1 minute? since we don't really care about the precision , to be able to recover more reminders in case of a downtime? 
Should we process all reminders? or keep this as it is

Reply:
Yeah I'm not too fussed about precision, 15 seconds is fine

The minute is fine

Reply:
Cool done , committing in few